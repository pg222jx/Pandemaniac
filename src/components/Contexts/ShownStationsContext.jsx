import React, { useState, createContext } from 'react'

export const ShownStationsContext = createContext()

export const ShownStationsProvider = props => {

    const [shownStations, setShownStations] = useState([])

    return (
        <ShownStationsContext.Provider value={[shownStations, setShownStations]}>
            {props.children}
        </ShownStationsContext.Provider>
    )
}

