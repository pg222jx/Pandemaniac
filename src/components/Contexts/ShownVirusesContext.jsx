import React, { useState, createContext } from 'react';

export const ShownVirusesContext = createContext()

export const ShownVirusesProvider = props => {

    const [shownViruses, setShownViruses] = useState([])

    return (
        <ShownVirusesContext.Provider value={[shownViruses, setShownViruses]}>
            {props.children}
        </ShownVirusesContext.Provider>
    )
}

