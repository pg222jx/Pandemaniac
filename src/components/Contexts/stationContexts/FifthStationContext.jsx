import React,{ useState, createContext } from 'react';

export const FifthStationContext = createContext()

export const FifthStationProvider = props => {

    const [style, setStyle] = useState({bottom: undefined, left: undefined, visibility: 'hidden'})

    return (
        <FifthStationContext.Provider value={[style, setStyle]}>
            {props.children}
        </FifthStationContext.Provider>
    );
}

