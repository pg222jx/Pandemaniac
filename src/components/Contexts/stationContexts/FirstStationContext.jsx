import React,{ useState, createContext } from 'react';

export const FirstStationContext = createContext()

export const FirstStationProvider = props => {

    const [style, setStyle] = useState({bottom: undefined, left: undefined, visibility: 'hidden'})

    return (
        <FirstStationContext.Provider value={[style, setStyle]}>
            {props.children}
        </FirstStationContext.Provider>
    );
}

