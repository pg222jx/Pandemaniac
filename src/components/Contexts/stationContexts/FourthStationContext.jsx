import React,{ useState, createContext } from 'react';

export const FourthStationContext = createContext()

export const FourthStationProvider = props => {

    const [style, setStyle] = useState({bottom: undefined, left: undefined, visibility: 'hidden'})

    return (
        <FourthStationContext.Provider value={[style, setStyle]}>
            {props.children}
        </FourthStationContext.Provider>
    );
}

