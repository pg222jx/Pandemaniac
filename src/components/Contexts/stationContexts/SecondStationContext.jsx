import React,{ useState, createContext } from 'react';

export const SecondStationContext = createContext()

export const SecondStationProvider = props => {

    const [style, setStyle] = useState({bottom: undefined, left: undefined, visibility: 'hidden'})

    return (
        <SecondStationContext.Provider value={[style, setStyle]}>
            {props.children}
        </SecondStationContext.Provider>
    );
}

