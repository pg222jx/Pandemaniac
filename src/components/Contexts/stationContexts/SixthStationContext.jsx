import React,{ useState, createContext } from 'react';

export const SixthStationContext = createContext()

export const SixthStationProvider = props => {

    const [style, setStyle] = useState({bottom: undefined, left: undefined, visibility: 'hidden'})

    return (
        <SixthStationContext.Provider value={[style, setStyle]}>
            {props.children}
        </SixthStationContext.Provider>
    );
}

