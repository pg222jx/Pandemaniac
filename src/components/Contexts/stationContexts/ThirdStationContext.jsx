import React,{ useState, createContext } from 'react';

export const ThirdStationContext = createContext()

export const ThirdStationProvider = props => {

    const [style, setStyle] = useState({bottom: undefined, left: undefined, visibility: 'hidden'})

    return (
        <ThirdStationContext.Provider value={[style, setStyle]}>
            {props.children}
        </ThirdStationContext.Provider>
    );
}

