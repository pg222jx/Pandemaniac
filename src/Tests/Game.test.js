import React from 'react'
import ReactDOM from 'react-dom'
import { render } from '@testing-library/react'
import { shallow } from 'enzyme'
import Game from '../components/Game'

import { ShownVirusesProvider } from '../components/Contexts/ShownVirusesContext'
import { ShownStationsProvider } from '../components/Contexts/ShownStationsContext'
import { TimerProvider } from '../components/Contexts/TimerContext'

import { FirstStationProvider } from '../components/Contexts/stationContexts/FirstStationContext'
import { SecondStationProvider } from '../components/Contexts/stationContexts/SecondStationContext'
import { ThirdStationProvider } from '../components/Contexts/stationContexts/ThirdStationContext'
import { FourthStationProvider } from '../components/Contexts/stationContexts/FourthStationContext'
import { FifthStationProvider } from '../components/Contexts/stationContexts/FifthStationContext'
import { SixthStationProvider } from '../components/Contexts/stationContexts/SixthStationContext'

const wrapperShallow = shallow(
<FirstStationProvider>
    <SecondStationProvider>
        <ThirdStationProvider>
            <FourthStationProvider>
                <FifthStationProvider>
                   <SixthStationProvider>
                        <ShownVirusesProvider>
                            <ShownStationsProvider>
                                <TimerProvider>
                                    <Game/>   
                                </TimerProvider>
                            </ShownStationsProvider>
                        </ShownVirusesProvider>
                    </SixthStationProvider>
                </FifthStationProvider>
            </FourthStationProvider>
        </ThirdStationProvider>
    </SecondStationProvider>
</FirstStationProvider>
)


describe('testing the Game component', () => {
    it('should match the snapshot', () => {
        expect(wrapperShallow.html()).toMatchSnapshot()
      })

      
    it('should render without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(
        <FirstStationProvider>
            <SecondStationProvider>
                <ThirdStationProvider>
                    <FourthStationProvider>
                        <FifthStationProvider>
                        <SixthStationProvider>
                                <ShownVirusesProvider>
                                    <ShownStationsProvider>
                                        <TimerProvider>
                                            <Game/>   
                                        </TimerProvider>
                                    </ShownStationsProvider>
                                </ShownVirusesProvider>
                            </SixthStationProvider>
                        </FifthStationProvider>
                    </FourthStationProvider>
                </ThirdStationProvider>
            </SecondStationProvider>
        </FirstStationProvider>, div
        )
    })
})
