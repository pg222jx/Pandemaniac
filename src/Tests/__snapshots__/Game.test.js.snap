// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the Game component should match the snapshot 1`] = `"<div class=\\"Game\\" data-testid=\\"GameTest\\"><div class=\\"navbar\\"><div class=\\"dropdown\\"><button aria-haspopup=\\"true\\" aria-expanded=\\"false\\" id=\\"rules\\" type=\\"button\\" class=\\"navLinks dropdown-toggle btn btn-primary\\">Rules</button></div><div class=\\"dropdown\\"><button aria-haspopup=\\"true\\" aria-expanded=\\"false\\" id=\\"controls\\" type=\\"button\\" class=\\"navLinks dropdown-toggle btn btn-primary\\">Controls</button></div><div class=\\"dropdown\\"><button aria-haspopup=\\"true\\" aria-expanded=\\"false\\" id=\\"highscore_title\\" type=\\"button\\" class=\\"navLinks dropdown-toggle btn btn-primary\\">Highscore</button></div></div><br/><br/><h1 class=\\"pandemaniacTitle\\">Pandemaniac</h1><div class=\\"gameContainer\\"><img src=\\"map.png\\" alt=\\"World map\\" class=\\"worldMap\\" useMap=\\"#mapImage\\"/><map name=\\"mapImage\\"><area href=\\"#\\" alt=\\"North America\\" shape=\\"rect\\" coords=\\"20,20, 400,240\\"/><area href=\\"#\\" alt=\\"South America\\" shape=\\"rect\\" coords=\\"155,255, 350,520\\"/><area href=\\"#\\" alt=\\"Europe\\" shape=\\"rect\\" coords=\\"435,20, 630,140\\"/><area href=\\"#\\" alt=\\"Asia\\" shape=\\"rect\\" coords=\\"640,20, 950,200\\"/><area href=\\"#\\" alt=\\"Africa\\" shape=\\"rect\\" coords=\\"435,170, 580,450\\"/><area href=\\"#\\" alt=\\"Australia\\" shape=\\"rect\\" coords=\\"800,300, 1100,480\\"/></map><div style=\\"visibility:hidden\\"><img src=\\"station.png\\" alt=\\"Science station for North America\\" id=\\"firstStation\\" class=\\"station\\" style=\\"visibility:hidden\\"/><img src=\\"station.png\\" alt=\\"Science station for South America\\" id=\\"secondStation\\" class=\\"station\\" style=\\"visibility:hidden\\"/><img src=\\"station.png\\" alt=\\"Science station for Africa\\" id=\\"thirdStation\\" class=\\"station\\" style=\\"visibility:hidden\\"/><img src=\\"station.png\\" alt=\\"Science station for Europe\\" id=\\"fourthStation\\" class=\\"station\\" style=\\"visibility:hidden\\"/><img src=\\"station.png\\" alt=\\"Science station for Asia\\" id=\\"firthStation\\" class=\\"station\\" style=\\"visibility:hidden\\"/><img src=\\"station.png\\" alt=\\"Science station for Australia\\" id=\\"sixthStation\\" class=\\"station\\" style=\\"visibility:hidden\\"/><img src=\\"player.png\\" alt=\\"The player\\" class=\\"player\\"/><div class=\\"virusContainer\\"><img src=\\"virus.png\\" alt=\\"Virus\\" class=\\"virus\\" style=\\"bottom:390px;left:120px;visibility:hidden\\" virusid=\\"NA 1\\"/><img src=\\"virus.png\\" alt=\\"Virus\\" class=\\"virus\\" style=\\"bottom:410px;left:140px;visibility:hidden\\" virusid=\\"NA 2\\"/><img src=\\"virus.png\\" alt=\\"Virus\\" class=\\"virus\\" style=\\"bottom:390px;left:140px;visibility:hidden\\" virusid=\\"NA 3\\"/><img src=\\"virus.png\\" alt=\\"Virus\\" class=\\"virus\\" style=\\"bottom:160px;left:245px;visibility:hidden\\" virusid=\\"SA 1\\"/><img src=\\"virus.png\\" alt=\\"Virus\\" class=\\"virus\\" style=\\"bottom:180px;left:265px;visibility:hidden\\" virusid=\\"SA 2\\"/><img src=\\"virus.png\\" alt=\\"Virus\\" class=\\"virus\\" style=\\"bottom:160px;left:265px;visibility:hidden\\" virusid=\\"SA 3\\"/><img src=\\"virus.png\\" alt=\\"Virus\\" class=\\"virus\\" style=\\"bottom:395px;left:535px;visibility:hidden\\" virusid=\\"EU 1\\"/><img src=\\"virus.png\\" alt=\\"Virus\\" class=\\"virus\\" style=\\"bottom:415px;left:555px;visibility:hidden\\" virusid=\\"EU 2\\"/><img src=\\"virus.png\\" alt=\\"Virus\\" class=\\"virus\\" style=\\"bottom:395px;left:555px;visibility:hidden\\" virusid=\\"EU 3\\"/><img src=\\"virus.png\\" alt=\\"Virus\\" class=\\"virus\\" style=\\"bottom:250px;left:500px;visibility:hidden\\" virusid=\\"AF 1\\"/><img src=\\"virus.png\\" alt=\\"Virus\\" class=\\"virus\\" style=\\"bottom:270px;left:520px;visibility:hidden\\" virusid=\\"AF 2\\"/><img src=\\"virus.png\\" alt=\\"Virus\\" class=\\"virus\\" style=\\"bottom:250px;left:520px;visibility:hidden\\" virusid=\\"AF 3\\"/><img src=\\"virus.png\\" alt=\\"Virus\\" class=\\"virus\\" style=\\"bottom:395px;left:735px;visibility:hidden\\" virusid=\\"AS 1\\"/><img src=\\"virus.png\\" alt=\\"Virus\\" class=\\"virus\\" style=\\"bottom:415px;left:755px;visibility:hidden\\" virusid=\\"AS 2\\"/><img src=\\"virus.png\\" alt=\\"Virus\\" class=\\"virus\\" style=\\"bottom:395px;left:755px;visibility:hidden\\" virusid=\\"AS 3\\"/><img src=\\"virus.png\\" alt=\\"Virus\\" class=\\"virus\\" style=\\"bottom:110px;left:890px;visibility:hidden\\" virusid=\\"AU 1\\"/><img src=\\"virus.png\\" alt=\\"Virus\\" class=\\"virus\\" style=\\"bottom:130px;left:910px;visibility:hidden\\" virusid=\\"AU 2\\"/><img src=\\"virus.png\\" alt=\\"Virus\\" class=\\"virus\\" style=\\"bottom:110px;left:910px;visibility:hidden\\" virusid=\\"AU 3\\"/></div><div class=\\"turnCard\\"><div class=\\"TurnsDone\\"><h4>Turns done:</h4><p>0</p><hr/></div><div class=\\"timerContainer\\"><h4>Time spent:</h4><p data-testid=\\"TimeSpent-test\\" class=\\"TimeSpent\\">0:0</p><hr/></div><div class=\\"turnContainer\\"><h4 class=\\"ActionsLeft\\">Actions left: 2</h4><hr/><h4>What&#x27;s your next move?</h4><button data-testid=\\"value\\" class=\\"Turn\\" id=\\"move\\">Move one step</button><button class=\\"Turn\\">Build a science centre</button><button class=\\"Turn\\" id=\\"remove\\">Remove virus</button><button class=\\"Turn\\" id=\\"done\\">Done with turn!</button></div></div></div><div class=\\"gameOverContainer\\"></div><div class=\\"playerWonContainer\\"></div><div class=\\"startGameContainer\\"><div class=\\"startButton\\" style=\\"visibility:visible\\"><button>Start Game!</button></div></div><div class=\\"informationContainer\\"><p class=\\"information\\">Don&#x27;t forget to read the rules before start!</p></div></div></div>"`;
